Engine/Entities/StateEntity.cs

```diff
@@ -20,7 +20,7 @@ public StateEntity(string name, Transform transform, World world, IState initial
         /// <summary>
         /// Current state of the Entity.
         /// </summary>
-        protected IState State { get; set; }
+        public IState State { get; protected set; }
```

Comment:
Maybe remove ChangeState and use the setter => create ticket.

Change to protected